envsec/internal/filecache/filecache.go

```diff
@@ -68,7 +65,11 @@ func (c *cache) Get(key string) ([]byte, error) {
 }
 
 func (c *cache) filename(key string) string {
-	dir := xdg.CacheSubpath(c.appName)
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		cacheDir = "~/.cache"
+	}
+	dir := filepath.Join(cacheDir, c.domain)
 	_ = os.MkdirAll(dir, 0755)
-	return xdg.CacheSubpath(filepath.Join(c.appName, prefix+key))
+	return filepath.Join(dir, key)
```

Comment:
Re: CICD failure: Could it be this line `os.MkdirAll(dir, 0755)` creating the `~/.cache` dir under a specific user?